policyengine_us/variables/gov/states/hi/tax/income/credits/cdcc/dependent_care_benefit/hi_min_head_spouse_earned.py

```diff
@@ -0,0 +1,71 @@
+from policyengine_us.model_api import *
+
+
+class hi_min_head_spouse_earned(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Hawaii minimum income between head and spouse for the CDCC"
+    defined_for = StateCode.HI
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://files.hawaii.gov/tax/forms/2022/n11ins.pdf#page=28"
+        "https://files.hawaii.gov/tax/forms/2022/n11ins.pdf#page=29"
+        "https://files.hawaii.gov/tax/legal/hrs/hrs_235.pdf#page=41"
+        "https://files.hawaii.gov/tax/forms/2022/schx_i.pdf#page=2"
+    )
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.hi.tax.income.credits.cdcc
+        person = tax_unit.members
+        head = person("is_tax_unit_head", period)
+        spouse = person("is_tax_unit_spouse", period)
+        head_or_spouse = head | spouse
+        # Head or spouse are eligible for an income floor if disabled or a student
+        eligible = head_or_spouse & (
+            person("is_disabled", period)
+            | person("is_full_time_student", period)
+        )
+        qualified_children = tax_unit("count_cdcc_eligible", period)
+
+        # Floor depends on number of eligible dependents
+        income_floor = select(
+            [
+                qualified_children <= 1,
+                qualified_children > 1,
+            ],
+            [
+                p.dependent_care_benefits.expense_floor.one_child,
+                p.dependent_care_benefits.expense_floor.two_or_more_child,
+            ],
+        )
+        income = person("earned_income", period)
+        head_or_spouse_income = head_or_spouse * income
+        increased_income = max_(head_or_spouse_income, income_floor)
+        uncapped_income = where(
+            eligible,
+            increased_income,
+            head_or_spouse_income,
+        )
+        # remove impact of 0 income output for dependents
+        head_spouse_income = where(
+            head_or_spouse, uncapped_income, tax_unit.max(uncapped_income)
+        )
+        # Edge case: both spouses were students or disabled:
+        # compare original income with eligible income
+        both_disabled_income = where(
```

Comment:
I think I understand why this works but I think restructuring it by person would make it more readable. How about creating a new person level variable for the relevant earnings. To capture the student disabled logic, compare their earnings to their spouse's and apply the floor unless they're both eligible and their earnings are less (or take the head's if a tie).

Reply:
I now created a new person-level variable called “hi_eligible_income_floor” to obtain the income floor according to the number of dependents and return the increased income. I try to make “hi_min_head_spouse_earned” more readable and concise and also avoid having duplicate codes in two formulas. Please tell me if you have any other thoughts.